src/crypto-providers/ledgerCryptoProvider.ts

```diff
@@ -106,13 +121,18 @@ export const LedgerCryptoProvider: () => Promise<CryptoProvider> = async () => {
     }
   }
 
-  const determineSigningMode = (certificates: _Certificate[], signingFiles: HwSigningData[]) => {
-    const poolRegistrationCert = certificates.find(
+  const determineSigningMode = (
+    unsignedTxParsed: _UnsignedTxParsed,
+    signingFiles: HwSigningData[],
+  ) => {
+    const poolRegistrationCert = unsignedTxParsed.certificates.find(
       (cert) => cert.type === TxCertificateKeys.STAKEPOOL_REGISTRATION,
     ) as _StakepoolRegistrationCert
 
     if (!poolRegistrationCert) {
```

Comment:
I'd refactor this to completely deal with poolRegistrationCert first (in this if) and all the other signing modes only after that.